src/components/Reports/GoalCalculator/SharedComponents/GoalCalculatorGrid/GoalCalculatorGrid.test.tsx

```diff
@@ -1,122 +1,196 @@
+    // Check that the UpdatePrimaryBudgetCategory mutation was called
     await waitFor(() => {
-      expect(getByText('Total')).toBeInTheDocument();
-      expect(getByText('$5,500')).toBeInTheDocument();
+      expect(mutationSpy).toHaveGraphqlOperation(
+        'UpdatePrimaryBudgetCategory',
+        {
+          input: {
+            accountListId: 'account-list-1',
+            directInput: 1500,
+          },
+        },
+      );
     });
   });
 
   it('adds a new row when Add button is clicked', async () => {
-    const { getByRole, getByText } = render(
+    const { getByRole, findByText } = render(
       <GoalCalculatorTestWrapper>
-        <GoalCalculatorGrid {...defaultProps} />
+        <TestComponent />
       </GoalCalculatorTestWrapper>,
     );
+    const lumpSumButton = await findByText('Line Item');
+    userEvent.click(lumpSumButton);
 
-    userEvent.click(
-      getByRole('button', {
-        name: /Add Line Item/i,
-      }),
+    const addButton = getByRole('button', { name: /Add Line Item/i });
+    userEvent.click(addButton);
+    expect(await findByText('New Income')).toBeInTheDocument();
+  });
+
+  it("doesn't remove base rows with categories", async () => {
+    const { getByText, findByText } = render(
+      <GoalCalculatorTestWrapper>
+        <TestComponent />
+      </GoalCalculatorTestWrapper>,
     );
+    const lumpSumButton = await findByText('Line Item');
+    userEvent.click(lumpSumButton);
 
-    expect(getByText('New Income')).toBeInTheDocument();
-    expect(getByText('$5,500')).toBeInTheDocument();
+    expect(getByText('Compass Room')).toBeInTheDocument();
+    expect(getByText('Desk')).toBeInTheDocument();
+    expect(getByText('Eyes PaintBrush Bible')).toBeInTheDocument();
+
+    const compassRow = getByText('Compass Room').closest('[role="row"]');
+    userEvent.hover(compassRow!);
+    const deleteButton = compassRow?.querySelector('[aria-label="Delete"]');
+
+    expect(deleteButton).not.toBeInTheDocument();
   });
 
-  it('removes a row when delete button is clicked', async () => {
-    const { getByText, queryByText } = render(
+  it('adds and removes a row when delete button is clicked', async () => {
+    const { getByText, findByText } = render(
       <GoalCalculatorTestWrapper>
-        <GoalCalculatorGrid {...defaultProps} />
+        <TestComponent />
       </GoalCalculatorTestWrapper>,
     );
-    const freelanceRow = getByText('Freelance Work').closest('[role="row"]');
-    userEvent.hover(freelanceRow!);
-    const deleteButton = freelanceRow?.querySelector('[aria-label="Delete"]');
-    if (deleteButton) {
-      userEvent.click(deleteButton as Element);
-    }
+    const lumpSumButton = await findByText('Line Item');
+    userEvent.click(lumpSumButton);
+
+    userEvent.click(getByText(/Add Line Item/i));
+    expect(await findByText('New Income')).toBeInTheDocument();
+    const newIncomeRow = getByText('New Income').closest('[role="row"]');
```

Comment:
`.toBeInTheDocument()` checks don't really do anything and are only for when we're not doing anything else with the found element.

```suggestion
    const newIncomeRow = (await findByText('New Income')).closest('[role="row"]');
```